### PR TITLE
Follow up custom data

### PR DESCRIPTION
**Pull request by laradicp**

### CODE REVIEW

File: src/callbacks.jl [hunk 1]
```diff
@@ -90,3 +90,19 @@ function callback_ub(cbdata, x::JuMP.VariableRef)
         PricingVariableUpperBound(cbdata), index(x)
     )
 end
+
+"""
+    AbstractCustomData
+
+Every custom data assigned to a pricing variable or a user cut should inherit from it.
+"""
+abstract type AbstractCustomData end
```
Comment (guimarqu) [inline src/callbacks.jl:99]: Does it replace `ColunaBase.AbstractCustomData` ?

File: src/customdata.jl [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    model, CustomVars(), customvars
+)
+
+addcustomconstrs!(model, customconstrs::Vector{DataType}) = MOI.set(
+    model, CustomConstrs(), customconstrs
+)
+
+customvars(model) = MOI.get(model, CustomVars())
+
+customconstrs(model) = MOI.get(model, CustomConstrs())
```
Comment (guimarqu) [inline src/customdata.jl:14]: Can you add a docstring for these four methods ?

File: src/callbacks.jl (hunk 1 above)
Comment (laradicp) [reply src/callbacks.jl:99]: This change was from @artalvpes commit. I forgot to ask for his review too.

File: src/customdata.jl (hunk 2 above)
Comment (laradicp) [reply src/customdata.jl:14]: Should I keep the names `addcustomvars!`/`addcustomconstrs!` and `customvars`/`customconstrs`? I added `customvars`/`customconstrs`, but for me something like `setcustomvars!`/`setcustomconstrs!` and `getcustomvars`/`getcustomconstrs` would maybe describe better these methods.

Comment (guimarqu) [reply src/customdata.jl:34]: I think `customvars` and `customvars!` are ok to keep consistency with other method names.

File: src/BlockDecomposition.jl [hunk 3]
```diff
@@ -9,7 +9,8 @@ const MOIU = MOI.Utilities
 const JC = JuMP.Containers
 
 export BlockModel, annotation, specify!, gettree, getmaster, getsubproblems, ×, indice,
-       objectiveprimalbound!, objectivedualbound!, branchingpriority!, branchingpriority
+       objectiveprimalbound!, objectivedualbound!, branchingpriority!, branchingpriority,
+       addcustomvars!, addcustomconstrs!, customvars, customconstrs
```
Comment (guimarqu) [inline src/BlockDecomposition.jl:13]:
```suggestion
       customvars!, customconstrs!, customvars, customconstrs
```

File: src/customdata.jl [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+)
+
+"""
+    addcustomconstrs!(model, customconstrs::Vector{DataType})
+
+Set the types of custom constraints in a model.
+"""
+addcustomconstrs!(model, customconstrs::Vector{DataType}) = MOI.set(
+    model, CustomConstrs(), customconstrs
+)
```
Comment (guimarqu) [inline src/customdata.jl:20]:
```suggestion
"""
    customvars!(model, customvars::Vector{DataType})

Set the possible custom data types of variables in a model.
"""
customvars!(model, customvars::Vector{DataType}) = MOI.set(
    model, CustomVars(), customvars
)

"""
    customconstrs!(model, customconstrs::Vector{DataType})

Set the possible custom data types of constraints in a model.
"""
customconstrs!(model, customconstrs::Vector{DataType}) = MOI.set(
    model, CustomConstrs(), customconstrs
)
```

File: src/customdata.jl [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+"""
+    customvars(model)
+
+Return the types of custom variables in a model.
```
Comment (guimarqu) [inline src/customdata.jl:25]:
```suggestion
Return the possible custom data types of variables in a model.
```

File: src/customdata.jl [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+"""
+customvars(model) = MOI.get(model, CustomVars())
+
+"""
+    customconstrs(model)
+
+Return the types of custom constraints in a model.
```
Comment (guimarqu) [inline src/customdata.jl:32]:
```suggestion
Return the custom data types of constraints in a model.
```

File: test/customdata.jl [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+end
+
+struct MyCustomCutData2 <: BlockDecomposition.AbstractCustomData
+    min_items::Float64
+end
+
+function test_custom_data()
+    model = Model()
+    addcustomvars!(model, [MyCustomVarData1, MyCustomVarData2])
+    addcustomconstrs!(model, [MyCustomCutData1, MyCustomCutData2])
```
Comment (guimarqu) [inline test/customdata.jl:20]:
```suggestion
    customvars!(model, [MyCustomVarData1, MyCustomVarData2])
    customconstrs!(model, [MyCustomCutData1, MyCustomCutData2])
```

**Review by guimarqu — COMMENTED** [review]
rename `addcustomvars!` methods and change docstrings

File: src/callbacks.jl (hunk 1 above)
Comment (artalvpes) [reply src/callbacks.jl:99]: When I added it, I was not aware of ```ColunaBase.AbstractCustomData```. You can remove it.

**Review by guimarqu — APPROVED** [review]

**Pull request merged**